config/account.go

```diff
@@ -297,3 +303,36 @@ func (a *AccountChannel) IsSet() bool {
 type AccountPrivacy struct {
 	AllowActivities AllowActivities `mapstructure:"allowactivities" json:"allowactivities"`
 }
+
+type IpMasking struct {
+	IpV6 IpMasks `mapstructure:"ipv6" json:"ipv6"`
+	IpV4 IpMasks `mapstructure:"ipv4" json:"ipv4"`
+}
+
+type IpMasks struct {
+	ActivityLeftMaskBits    int `mapstructure:"activity_left_mask_bits" json:"activity_left_mask_bits"`
+	GdprLeftMaskBitsLowest  int `mapstructure:"gdpr_left_mask_bits_lowest" json:"gdpr_left_mask_bits_lowest"`
+	GdprLeftMaskBitsHighest int `mapstructure:"gdpr_left_mask_bits_highest" json:"gdpr_left_mask_bits_highest"`
+}
+
+func (ipMasking *IpMasking) Validate(errs []error) []error {
+	errs = ipMasking.IpV6.validate(errs, Ipv6Bits, "ipv6")
```

Comment:
You need to handle the case of `ipMasking` being nil.

Reply:
Technically `ipMasking` is never nil here. Should I modify it to a non-pointer?

Reply:
Yes, please. Let's add some defensive coding since `nil` is allowed here. Alternatively, you may be able to change this to a value type receiver.